Index: src/components/form/editableButtons.ts
===================================================================
--- src/components/form/editableButtons.ts	(revision 7e25aa5f07b2a453155b311a1086caa8db43269a)
+++ src/components/form/editableEntityButtons.ts	
@@ -1,12 +1,12 @@
 export const mod = angular.module('iow.components.form');
 
-mod.directive('editableButtons', () => {
+mod.directive('editableEntityButtons', () => {
   return {
     restrict: 'E',
     scope: {
       ctrl: '=editableController'
     },
-    template: require('./editableButtons.html'),
+    template: require('./editableEntityButtons.html'),
     transclude: true
   };
 });
